engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java

```diff
@@ -277,7 +287,7 @@ public Object invokeWarnings(
                                     invokeFunctionNode.getSchema(),
                                     invokeFunctionNode.getDefaultsExecutionMode(),
                                     invokeFunctionNode.getArgumentsExecutionMode()));
-            childDispatch.setTailStatus(TailStatus.NOT_TAIL);
+            childDispatch.setTailStatus(getTailStatus());
```

Comment:
In the [test below](https://github.com/enso-org/enso/pull/7116/files#r1240600638)... what node behavior this change affects?

The warning in the test is attached to an argument, not to `self`, so I don't see how this change can be triggered?

Reply:
Oh you mean why I change it in `InvokeCallableNode` when the fix was only necessary in `InvokeMethodNode`, right? 
I'm reverting the previous change for all three cases and replacing it with the same approach i.e. warnings propagation via `TailCallException`.
It does not appear that we have any test case  exercising this case specifically.